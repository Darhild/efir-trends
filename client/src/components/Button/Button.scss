@import './../../variables';

.Button {
    display: flex;
    justify-content: center;
    align-items: center;
    box-sizing: border-box;
    text-align: center;
    color: inherit;
    text-decoration: none;
    background: var(--color-accent);
    cursor: pointer;

    &_type_addVideo {
        position: fixed;
        right: calc(50% - 87px);
        bottom: 22px;
        z-index: 2;
        width: 174px;
        padding: 10px 0;
        font-size: 14px;
        line-height: 20px;
        border-radius: 100px;

        @media(min-width: $screen-s) {
            position: absolute;
            top: 178px;
            right: 142px;
            width: 150px;
            height: 36px;
            border-radius: 4px;
        }
    }

    &_type_subscribe {
        position: absolute;
        top: 148px;
        width: 100%;
        padding: 10px;
        font-size: 13px;
        line-height: 16px;
        border-radius: 4px;

        @media(min-width: $screen-s) {
            top: 178px;
            right: 0;
            width: 134px;
        }
    }

    &_subscribe_off {
        background: var(--color-background-button);
    }

    &-Icon {
        margin-right: 6px;
    }
}