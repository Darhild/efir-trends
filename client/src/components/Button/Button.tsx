import React from 'react';
import classnames from 'classnames';
import { ReactComponent as Checkmark } from '../../images/svg/checkmark.svg';
import { ReactComponent as Camera } from '../../images/svg/camera.svg';
import './Button.scss';

interface ButtonProps {
    type: string;
}

interface SubscribeButtonState {
    isSubscribed?: boolean;
}

interface TypedButtonProps {
    cn: string;
}

class SubscribeButton extends React.Component<TypedButtonProps, SubscribeButtonState> {
    public state = {
        isSubscribed: false,
    };

    public subscribe = () => {
        this.setState((state) => ({
            isSubscribed: !state.isSubscribed,
        }));
    }

    public render = () => {
        const { cn } = this.props;
        const { isSubscribed } = this.state;
        const subscribeButtonCn = classnames(
            cn,
            isSubscribed && 'Button_subscribe_off',
        );

        return (
            <div className={subscribeButtonCn} onClick={this.subscribe}>
                {isSubscribed
                    ? <>
                        <Checkmark className="Button-Icon"/>
                        <span>Вы подписаны</span>
                    </>
                    : <span>Подписаться</span>}
            </div>
        );
    }
}

const AddVideoButton = ({ cn }: TypedButtonProps) => (
    <div className={cn}>
        <Camera className="Button-Icon"/>
        <span>Добавить видео</span>
    </div>
);

const Button = ({ type }: ButtonProps) => {
        const buttonCn = classnames('Button', `Button_type_${type}`);

        return (
            type === 'subscribe'
                ? <SubscribeButton cn={buttonCn} />
                : <AddVideoButton cn={buttonCn} />
        );
};

export default Button;
