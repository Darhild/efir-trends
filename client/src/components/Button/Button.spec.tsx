import React from 'react';
import Button from './Button';
import renderer from 'react-test-renderer';

test('subscribe Button default state', () => {
    const component = renderer.create(<Button type="subscribe" />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
});
