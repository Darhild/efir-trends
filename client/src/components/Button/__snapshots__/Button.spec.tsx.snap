// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`subscribe Button default state 1`] = `
<div
  className="Button Button_type_subscribe"
  onClick={[Function]}
>
  <span>
    Подписаться
  </span>
</div>
`;
